library/src/main/java/io/appwrite/Client.kt

```diff
@@ -0,0 +1,246 @@
+package io.appwrite
+
+import android.content.Context
+import com.franmontiel.persistentcookiejar.ClearableCookieJar
+import com.franmontiel.persistentcookiejar.PersistentCookieJar
+import com.franmontiel.persistentcookiejar.cache.SetCookieCache
+import com.franmontiel.persistentcookiejar.persistence.SharedPrefsCookiePersistor
+import com.google.gson.Gson
+import io.appwrite.exceptions.AppwriteException
+import io.appwrite.extensions.JsonExtensions.fromJson
+import io.appwrite.models.Error
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.suspendCancellableCoroutine
+import okhttp3.*
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.RequestBody.Companion.asRequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.logging.HttpLoggingInterceptor
+import java.io.BufferedReader
+import java.io.File
+import java.io.IOException
+import java.security.SecureRandom
+import java.security.cert.X509Certificate
+import javax.net.ssl.HostnameVerifier
+import javax.net.ssl.SSLContext
+import javax.net.ssl.SSLSocketFactory
+import javax.net.ssl.TrustManager
+import javax.net.ssl.X509TrustManager
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+class Client(
+    context: Context,
+    var endPoint: String = "https://appwrite.io/v1",
+    private var selfSigned: Boolean = false
+) : CoroutineScope {
+
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Main + job
+
+    private val job = Job()
+
+    private lateinit var http: OkHttpClient
+
+    private val headers: MutableMap<String, String>
+    
+    val config: MutableMap<String, String>
+
+    val cookieJar: ClearableCookieJar = PersistentCookieJar(
+        SetCookieCache(), 
+        SharedPrefsCookiePersistor(context)
+    )
+
+    init {
+        headers = mutableMapOf(
+            "content-type" to "application/json",
+            "x-sdk-version" to "appwrite:kotlin:0.0.0-SNAPSHOT",            
+            "x-appwrite-response-format" to "0.7.0"
+        )
+        config = mutableMapOf()
+        
+        setSelfSigned(selfSigned)
+    }
+
+    /// Your project ID
+    fun setProject(value: String): Client {
+        config["project"] = value
+        addHeader("X-Appwrite-Project", value)
+        return this
+    }
+
+    /// Your secret JSON Web Token
+    fun setJWT(value: String): Client {
+        config["jWT"] = value
```

Comment:
Umm `jWT`?

Reply:
Leaving it like this for now- as we discussed